qiskit_aer/backends/aer_compiler.py

```diff
@@ -328,6 +347,95 @@ def _inline_if_else_op(self, instruction, continue_label, break_label, parent, b
 
         parent.append(AerMark(if_end_label, len(qargs), len(mark_cargs)), qargs, mark_cargs)
 
+    def _inline_switch_case_op(self, instruction, continue_label, break_label, parent, bit_map):
+        """inline switch cases with jump and mark instructions"""
+        cases = instruction.operation.cases_specifier()
+
+        self._last_flow_id += 1
+        switch_id = self._last_flow_id
+        switch_name = f"switch_{switch_id}"
+
+        qargs = [bit_map[q] for q in instruction.qubits]
+        cargs = [bit_map[c] for c in instruction.clbits]
+        mark_cargs = cargs.copy()
+        mark_cargs.extend(
+            bit_map[c]
+            for c in (
+                (
+                    {instruction.operation.target}
+                    if isinstance(instruction.operation.target, Clbit)
+                    else set(instruction.operation.target)
+                )
+                - set(instruction.clbits)
+            )
+        )
+        case_labels = []
+        case_args_lists = []
+        case_bit_maps = []
+        case_bodies = []
```

Comment:
These four lists are hard for me to reason about, and I think they get out-of-sync; if `CASE_DEFAULT` is in a set of labels, we don't append anything to `case_args_lists`.  It might be simpler to make a single list whose scalar type is something like a
```python
case_object = collections.NamedTuple("case_object", ("label", "args", "bit_map", "body"))
```